tests/infrastructure/src/main/java/org/mule/test/infrastructure/process/Controller.java

```diff
@@ -69,19 +72,30 @@ public Controller(String muleHome, int timeout) {
     this.domainsDir = new File(muleHome + "/domains");
     this.appsDir = new File(muleHome + "/apps/");
     this.libsDir = new File(muleHome + "/lib/user");
+    this.internalRepository = new File(muleHome, "repository");
     this.wrapperConf = Paths.get(muleHome + "/conf/wrapper.conf");
     this.timeout = timeout != 0 ? timeout : DEFAULT_TIMEOUT;
   }
 
   public abstract String getMuleBin();
 
   public void start(String[] args) {
+    checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(args);
     int error = runSync("start", args);
     if (error != 0) {
       throw new MuleControllerException("The mule instance couldn't be started");
     }
   }
 
+  protected void checkRepositoryLocationAndUpdateInternalRepoPropertyIfPresent(String... args) {
+    Optional<String> repoVar =
+        Arrays.asList(args).stream().filter(arg -> arg.contains("-M-DmuleRuntimeConfig.maven.repositoryLocation=")).findFirst();
+    if (repoVar.isPresent()) {
+      this.internalRepository = new File(repoVar.get().split("=")[1]);
```

Comment:
The property repositoryLocation is to use another path for the repo of the distro. So in place of use the repository folder inside the distro you can use for example the /user/.m2/repository folder.